### PR TITLE
KVAC integration

### PR DESCRIPTION
**Pull request by lovesh**

### CODE REVIEW

File: tests/integration/anoncreds/wallet-util.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  const pk = privateKeyHex || randomAsHex(32);
+  const kp = generateEcdsaSecp256k1Keypair(pk);
+  kp.pk = pk;
+  return kp;
+}
+const docToKeyMap = {
+  Sr25519VerificationKey2020: 'sr25519',
+  Ed25519VerificationKey2018: 'ed25519',
+};
+export function keyDocToKeypair(keyDoc, dock) {
```
Comment (cykoder) [inline tests/integration/anoncreds/wallet-util.js:17]: minor formatting, missing new line

File: src/utils/inheritance.js [hunk 2]
```diff
@@ -33,21 +33,15 @@ export function withExtendedStaticProperties(properties, parentClass) {
...
-            `Static property \`${property}\` of \`${this.name}\` isn't extended properly`,
-          );
-        }
-
-        return value;
+        return this[propertySymbol];
       },
       set(newValue) {
-        if (newValue == null) {
+        /* if (newValue == null) {
```
Comment (cykoder) [inline src/utils/inheritance.js:39]: why commented? can it just be removed?

File: src/utils/vc/crypto/common/DockCryptoSignatureProof.js [hunk 3]
```diff
@@ -102,7 +103,6 @@ export default withExtendedStaticProperties(
         type,
         credentialSchema,
         credentialStatus,
-        issuer: _issuer,
```
Comment (cykoder) [inline src/utils/vc/crypto/common/DockCryptoSignatureProof.js:105]: why was this removed? could it make older credentials incompatible or something?

File: src/utils/vc/crypto/common/DockCryptoSignatureProof.js [hunk 4]
```diff
@@ -133,7 +133,7 @@ export default withExtendedStaticProperties(
         c.status = credentialStatus;
       }
       return {
-        version: proof.version,
+        version: proof.presVersion,
```
Comment (cykoder) [inline src/utils/vc/crypto/common/DockCryptoSignatureProof.js:136]: similar comment here, is this compatible with old credentials/presenting?

**Review by cykoder — DISMISSED** [review]
lgtm mostly just a couple formatting comments and clarifications

File: src/utils/vc/crypto/common/DockCryptoSignatureProof.js (hunk 3 above)
Comment (lovesh) [reply src/utils/vc/crypto/common/DockCryptoSignatureProof.js:105]: Good point. It can cause issue with verifying older presentations but not credentials. Fixed.

File: tests/integration/anoncreds/wallet-util.js (hunk 1 above)
Comment (lovesh) [reply tests/integration/anoncreds/wallet-util.js:19]: Fixed

File: src/utils/inheritance.js (hunk 2 above)
Comment (lovesh) [reply src/utils/inheritance.js]: Fixed

File: src/utils/vc/crypto/common/DockCryptoSignatureProof.js (hunk 4 above)
Comment (lovesh) [reply src/utils/vc/crypto/common/DockCryptoSignatureProof.js]: Similar to above, can cause issue older presentations but not credentials. Fixed.

File: src/utils/vc/crypto/common/DockCryptoKeyPair.js [hunk 5]
```diff
@@ -37,6 +37,10 @@ export default withExtendedStaticProperties(
       return new this(options);
     }
 
+    static paramGenerator() {
```
Comment (olegnn) [inline src/utils/vc/crypto/common/DockCryptoKeyPair.js:40]: Maybe would be better to define this function as
```
class DockCryptoKeyPair {
   ...
   static get generateParams() { return this.SignatureParams.getSigParamsOfRequiredSize; }
   ...
}
```

File: src/utils/vc/crypto/Bls12381BDDT16MACProofDock2024.js [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+      options,
+      Bls12381BDDT16MacProofDockName,
+      Bls12381BDDT16KeyPairDock2024,
+      'https://ld.dock.io/security/bddt16/v1',
+    );
+  }
+
+  async getVerificationMethod(_) {
+    // Public key isn't used for verifying KVACs
+    return Promise.resolve(undefined);
```
[the remark this answers is not in the document]
Comment (lovesh) [reply src/utils/vc/crypto/Bls12381BDDT16MACProofDock2024.js]: It was giving a lint error, suppressed it now.

File: src/utils/vc/crypto/common/DockCryptoKeyPair.js (hunk 5 above)
Comment (lovesh) [reply src/utils/vc/crypto/common/DockCryptoKeyPair.js]: Yes, this is cleaner. Fixed.

**Review by cykoder — APPROVED** [review]
